### ISSUE TITLE
Remove hardcoding of Patroller/Newbie roles

### ISSUE THREAD

**Issue by akinsey** [epochtalk/epochtalk#269]
**Problems caused by patroller/newbie code**
The main issue with the hardcoding of the `patroller` and `newbie` roles is that they are both bitcointalk specific features. If you checkout epochtalk and choose not to use the `bct-activity` module, the functionality of the `newbie` role breaks. The user will forever be a newbie and have their images stripped because they never reach the activity level that removes the `newbie` role as the `bct-activity` module isn't in use.

**Patroller References**
```
/bct-trust/permissions/index.js
/epochtalk/modules/ept-ads/permissions/index.js
/epochtalk/modules/ept-auto-moderation/permissions/index.js
/epochtalk/modules/ept-boards/permissions/index.js
/epochtalk/modules/ept-categories/permissions/index.js
/epochtalk/modules/ept-frontend/client/layout/header.admin.html
/epochtalk/modules/ept-frontend/client/layout/header.controller.js
/epochtalk/modules/ept-frontend/client/layout/header.html
/epochtalk/modules/ept-mentions/permissions/index.js
/epochtalk/modules/ept-messages/permissions/conversations.js
/epochtalk/modules/ept-messages/permissions/messages.js
/epochtalk/modules/ept-portal/permissions/index.js
/epochtalk/modules/ept-posts/authorization/delete.js
/epochtalk/modules/ept-posts/authorization/lock.js
/epochtalk/modules/ept-posts/authorization/update.js
/epochtalk/modules/ept-posts/common/index.js:
/epochtalk/modules/ept-posts/permissions/index.js
/epochtalk/modules/ept-reports/permissions/index.js
/epochtalk/modules/ept-threads/permissions/index.js
/epochtalk/modules/ept-users/permissions/index.js
/epochtalk/modules/ept-watchlist/permissions/index.js
/epochtalk/server/index.js
/epochtalk/server/plugins/acls/roles.js
/epochtalk/server/plugins/patroller/index.js
/epochtalk/server/plugins/patroller/routes.js
```

**Newbie References**
```
/bct-activity/hooks/index.js
/bct-trust/permissions/index.js
/epochtalk/modules/ept-ads/permissions/index.js
/epochtalk/modules/ept-auth/routes/invite-register.js
/epochtalk/modules/ept-auth/routes/register.js
/epochtalk/modules/ept-auto-moderation/permissions/index.js
/epochtalk/modules/ept-boards/permissions/index.js
/epochtalk/modules/ept-categories/permissions/index.js
/ept-mentions/permissions/index.js
/epochtalk/modules/ept-messages/permissions/conversations.js
epochtalk/modules/ept-messages/permissions/messages.js
/epochtalk/modules/ept-portal/permissions/index.js
/epochtalk/modules/ept-posts/authorization/delete.js
/epochtalk/modules/ept-posts/authorization/lock.js
/epochtalk/modules/ept-posts/authorization/update.js
/epochtalk/modules/ept-posts/common/index.js
/epochtalk/modules/ept-posts/permissions/index.js
/epochtalk/modules/ept-posts/routes/create.js
/epochtalk/modules/ept-posts/routes/update.js
/epochtalk/modules/ept-reports/permissions/index.js
/epochtalk/modules/ept-threads/permissions/index.js
/epochtalk/modules/ept-users/permissions/index.js
/epochtalk/modules/ept-watchlist/permissions/index.js
/epochtalk/server/plugins/acls/roles.js
/epochtalk/server/plugins/backoff/db.js [ Unused code DELETE ]
/epochtalk/server/plugins/patroller/db.js
```

**Problems to solve**
1. Create a hardcoded role inside of a module for Newbie and Patroller
2. Handle permissions for Newbie and Patroller for all other modules (ept-users, ept-threads, etc…) within the bct-patroller module. 
3. Is it okay to reference the newbie role within bct-activity hooks since they are both bct modules?
4. How do we allow authorization (eps-posts) methods to reference the patroller role which is defined in another module?
5. Turn hardcoded html referencing patroller code into directives within bct-patroller

**Comment by unenglishable** [comment]
👍 issue looks good

**Comment by akinsey** [comment]
Temporarily patched in #276

**Comment by akinsey** [comment]
Closing until this becomes relevant again
